File: benchmarks/fibonacci/java/Fib.java
public class Fib {
    static long fib(long n) {
        if (n < 2) return n;
        return fib(n - 1) + fib(n - 2);
    }

    public static void main(String[] args) {
        long n = args.length > 0 ? Long.parseLong(args[0]) : 32;
        if (fib(10) != 55) {
            System.err.println("self-check failed: fib(10) != 55");
            System.exit(1);
        }
        System.out.println(fib(n));
    }
}

File: benchmarks/fibonacci/rust/fib.rs
use std::env;

fn fib(n: u64) -> u64 {
    if n < 2 { n } else { fib(n - 1) + fib(n - 2) }
}

fn main() {
    let n: u64 = env::args().nth(1).map(|s| s.parse().unwrap()).unwrap_or(32);
    assert_eq!(fib(10), 55, "self-check failed: fib(10) != 55");
    println!("{}", fib(n));
}
